llvm/unittests/ADT/FunctionRefTest.cpp

```diff
@@ -59,4 +59,12 @@ TEST(FunctionRefTest, SFINAE) {
   EXPECT_EQ("string", returns([] { return "hello"; }));
 }
 
+TEST(FunctionRefTest, Equality) {
+  function_ref<int()> X = [] { return 1; };
+  function_ref<int()> Y = [] { return 2; };
+  EXPECT_TRUE(!(X == Y));
```

Comment:
```suggestion
  EXPECT_FALSE(X == Y);
  EXPECT_TRUE(X != Y);
```

also test something like
```
const auto Lambda = []() { return 0 };
function_ref<int()> A(Lambda), B (Lambda);
EXPECT_TRUE(A == B);
EXPECT_FALSE(A != B);
```

I think != should be automatically available but if it isn't then it's fine, don't bother testing it
```

Reply:
The != is not synthesized from ==, so dropped those test and added the one you suggested.